Src/Support/Google.Apis.Auth/OAuth2/IExternalAccountCredential.cs

```diff
@@ -0,0 +1,81 @@
+﻿/*
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Google.Apis.Http;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>
+    /// Represents and external account credential. Used mostly for unifying different
+    /// external account credentials hierarchies.
+    /// </summary>
+    internal interface IExternalAccountCredential: IGoogleCredential,
+        IHttpExecuteInterceptor, IHttpUnsuccessfulResponseHandler
+    {
+        /// <summary>
+        /// The STS audience which contains the resource name for the
+        /// workload identity pool or the workforce pool
+        /// and the provider identifier in that pool.
+        /// </summary>
+        string Audience { get; }
+
+        /// <summary>
+        /// The STS subject token type based on the OAuth 2.0 token exchange spec.
+        /// </summary>
+        string SubjectTokenType { get; }
+
+        /// <summary>
+        /// The STS token exchange endpoint.
+        /// </summary>
+        string TokenUrl { get; }
```

Comment:
Does this need to be part of the interface? Same for the other URL properties.

Reply:
(Just one drive-by comment for now.) This *is* in the interface. It's contained within `IExternalAccountCredential`. Were you thinking of a different interface??